crypto/evp/evp_lib.c

```diff
@@ -81,8 +81,12 @@ int evp_cipher_param_to_asn1_ex(EVP_CIPHER_CTX *c, ASN1_TYPE *type,
                                 evp_cipher_aead_asn1_params *asn1_params)
 {
     int ret = -1;                /* Assume the worst */
-    const EVP_CIPHER *cipher = c->cipher;
+    const EVP_CIPHER *cipher;
 
+    if (c == NULL || c->cipher == NULL)
+        return -1;
```

Comment:
Perhaps `goto err` would be better here to benefit from the `ERR_raise` that occurs in the event of all other errors in this function.

Reply:
adjusted